hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/SyntheticTxnFactory.java

```diff
@@ -308,6 +310,7 @@ public TransactionBody.Builder createNonfungibleApproval(
         if (approveWrapper.isFungible()) {
             builder.addTokenAllowances(TokenAllowance.newBuilder()
                     .setTokenId(approveWrapper.tokenId())
+                    .setOwner(ownerId.toGrpcAccountId())
```

Comment:
In my opinion, we should have a state migration that generates synthetic transactions to correct the downstream systems. There is no way for mirror node to otherwise get the corrected data.

Reply:
Thanks @steven-sheehy, so for each approval in state, we would generate a synthetic `ApproveAllowance` with the `owner` field set; and mirror node would do an idempotent upsert?

Reply:
Updated https://github.com/hashgraph/hedera-services/issues/5608 with a task to do this, will complete after merging of work @MiroslavGatsanoga is doing in https://github.com/hashgraph/hedera-services/pull/5799 to improve migration record exports.